src/type.js

```diff
@@ -222,6 +226,10 @@ async function typeImpl(
           newEntry = `${prevValue}.${char}`
         }
 
+        if (currentElement().type === 'date' && charPosition === 9) {
```

Comment:
Hmmmmm.... This seems to be coded for a specific use case (9 characters) rather than being very general. Are we sure that we'll be firing all the right events with these changes?

Reply:
It's a first naive implementation just to make its behavior green. :) I tried to not call any attention with a draft PR, sorry for that.

The problem with the issue is: we just can type the entire value, the input doesn't accept any invalid date. I think it was the issue with the input types too (number with `.` for example).
As for the date type, I don't have any current value in the interim to keep it around it.

I don't know yet what are the correct events to be triggered here. I think we can dispatch all of them (numbers and dashes in the same order passed to `type` (the `text`). What do you think?

Thank you.

Reply:
I think it's fine to do something like this do long as we fire all the same events a user would fire as they type into the date input. You might test it out here to see what events should be fired (it's possible what you have is correct): https://codesandbox.io/s/user-event-playground-eo909

I'm mostly concerned about 9 characters hard coded here. Will it always be exactly 9 characters?

Reply:
The events triggered in this PR are a little different from the playground: the browser triggers `input` (together with a `change`) events as soon as I start to type a valid date:

My input is `dd/07/2020`, I just need to type the day:

- when I type 2: `02/07/2020`:
```
input#date[value="2020-07-02"] - input
input#date[value="2020-07-02"] - change

```
- when I type 2 again: `22/07/2020`:
```
input#date[value="2020-07-22"] - input
input#date[value="2020-07-22"] - change
```

Reply:
Together with the `input` event, the browsers trigger a `change` event as soon as the date is valid. A little odd compared to the others.

Reply:
Is this different for other locales? I think this might be ok, though maybe it would be more resilient to try something like this:

```javascript
function isValidValue(el, value) {
  const clone = el.cloneNode()
  clone.value = value
  return clone.value === value
}
```

If the value is valid then we know we don't need to do anything special. Thoughts?

Reply:
Sorry for the delay.

I think it should work to avoid the hardcoded `charPosition`, but we'll have to keep some var around to control the `value` (from `isValueValud`) since we cannot type the `char` (what we have inside the `runCallbacks`) in a date input. Is't make sense for you or I'm missing something?

As for the events, just to make it clear: date input just triggers an `input` event (plus a `change` event) on a valid date according to the user-event playground. It's different from other events, though it's easy to implement it. Let's for it?

Thanks.

Reply:
Yeah, try that out and we'll see what it looks like. Thanks!